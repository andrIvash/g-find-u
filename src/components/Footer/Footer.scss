@import '../../styles/abstracts/functions';

.footer {
    padding: rem(10) 0;
}