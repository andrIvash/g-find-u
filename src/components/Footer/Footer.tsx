import React from 'react';

import './Footer.scss';

export const Footer = () => {
    return (
        <div className='footer center' data-testid='Footer'>
            &copy; 2024
        </div>
    );
};