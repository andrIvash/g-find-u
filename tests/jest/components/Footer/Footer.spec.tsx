import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Footer from '../../../../src/components/Footer';

describe('Footer Component', () => {
    it('renders without crashing', () => {
        render(<Footer />)
        expect(screen.getByText(/2024/)).toBeInTheDocument();
    });

    it('contains correct copyright text', () => {
        render(<Footer />);
        const footerElement = screen.getByText(/2024/);
        expect(footerElement).toHaveTextContent('Â© 2024');
    });

    it('has the correct class', () => {
        render(<Footer />);
        const footerElement = screen.getByTestId('Footer');
        expect(footerElement).toHaveClass('footer center');
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Footer />);
        expect(asFragment()).toMatchSnapshot();
    });
});