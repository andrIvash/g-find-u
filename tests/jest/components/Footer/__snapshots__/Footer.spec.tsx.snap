// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="footer center"
    data-testid="Footer"
  >
    Â© 2024
  </div>
</DocumentFragment>
`;
